python_modules/dagster/dagster/_core/definitions/observable_asset.py

```diff
@@ -1,19 +1,29 @@
-from dagster._core.definitions.asset_spec import ObservableAssetSpec
-from dagster._core.definitions.decorators import asset
+from typing import Sequence
 
+from dagster._core.definitions.asset_spec import (
+    SYSTEM_METADATA_KEY_ASSET_VARIETAL,
+    AssetSpec,
+    AssetVarietal,
+    ObservableAssetSpec,
+)
+from dagster._core.definitions.decorators.asset_decorator import multi_asset
 
-def create_unexecutable_observable_assets_def(observable_asset_spec: ObservableAssetSpec):
-    # This function is to be used in the internals of repository definition to coerce
-    # an ObservableAssetSepc into an AssetsDefinition
-    # TODO: will make this use multi_asset later in the stack
-    @asset(
-        key=observable_asset_spec.key,
-        description=observable_asset_spec.description,
-        metadata=observable_asset_spec.metadata,
-        group_name=observable_asset_spec.group_name,
-        deps=[
-            dep.asset_key for dep in observable_asset_spec.deps
-        ],  # switch to not using .asset_key once jamie's diff lands
+
+def create_unexecutable_observable_assets_def(specs: Sequence[ObservableAssetSpec]):
+    @multi_asset(
```

Comment:
I think we should enforce all or none on this special bit in AssetsDefinition - I don't think it ever makes sense to be partial

Reply:
I'm not following the comment. Do you mean explicitly erroring on properties we do not supprt?

Reply:
I mean if `SYSTEM_METADATA_KEY_ASSET_VARIETAL: AssetVarietal.UNEXECUTABLE.value` is set on any specs it must be set on all specs. An AssetDefinition contains all or none `is_executable` asset nodes

Reply:
I mean that is the net result of the current code? It does that for every `AssetSpec` it creates.

Reply:
yea the idea is just to explicitly guard that current condition / assumption against future changes